src/main.rs

```diff
@@ -534,5 +534,9 @@ fn enum_name(msg: &Message, signal: &Signal) -> String {
 }
 
 fn enum_variant_name(x: &str) -> String {
-    x.to_camel_case()
+    if !x.starts_with(|c: char| c.is_ascii_alphabetic()) {
```

Comment:
I am wondering if there is a more generic way to address possible enum variant naming issues, like is there some kind of crate for this? This also affects method names etc. in general. I switched to [quote](https://crates.io/crates/quote) in dbcc which covers this but maybe there is a less intrusive solution.

Reply:
I was looking into another problem where `type` was generated and I was thinking about just checking against all Rust keywords + numeric and symbols.

```rust
// https://doc.rust-lang.org/stable/reference/keywords.html
const KEYWORDS: [&str; 52] = [
    "as",
    "break",
    "const",
    "continue",
    "crate",
    "else",
    "enum",
    "extern",
    "false",
    "fn",
    "for",
    "if",
    "impl",
    "in",
    "let",
    "loop",
    "match",
    "mod",
    "move",
    "mut",
    "pub",
    "ref",
    "return",
    "self",
    "Self",
    "static",
    "struct",
    "super",
    "trait",
    "true",
    "type",
    "unsafe",
    "use",
    "where",
    "while",
    "async",
    "await",
    "dyn",
    "abstract",
    "become",
    "box",
    "do",
    "final",
    "macro",
    "override",
    "priv",
    "typeof",
    "unsized",
    "virtual",
    "yield",
    "try",
    "union",
];

pub fn is_keyword(x: &str) -> bool {
    KEYWORDS.contains(&x)
}

```

Reply:
I think keeping it simple is fine for now. I wrote some other code gen tools using syn and quote and generating strings to then rustfmt them… but it wasn't paying off.